Document removal of IrDeserializer.IrLinkerExtension and related classes in API changes (#1430)
JetBrains/intellij-sdk-docs: reference_guide/api_changes_list_2025.md

`org.jetbrains.kotlin.KtFakeSourceElement` class renamed to `org.jetbrains.kotlin.KtFakePsiSourceElement`
: Update code usages.

`org.jetbrains.kotlin.ir.linkage.IrDeserializer.IrLinkerExtension` class removed

> : This class from Kotlin compiler was removed and should not be used anymore.

Reviewer: ```suggestion
: This class was removed from the Kotlin compiler and is no longer available.
```

I think this is enough.

`org.jetbrains.kotlin.ir.builders.TranslationPluginContext` class removed

> : This class from Kotlin compiler was removed and should not be used anymore.

Author: ```suggestion
: This class was removed from the Kotlin compiler and is no longer available. 
```
